l2geth/core/types/transaction_signing.go

```diff
@@ -91,7 +40,16 @@ type sigCache struct {
 
 // MakeSigner returns a Signer based on the given chain config and block number.
 func MakeSigner(config *params.ChainConfig, blockNumber *big.Int) Signer {
-	return NewOVMSigner(config.ChainID)
+	var signer Signer
+	switch {
+	case config.IsEIP155(blockNumber):
+		signer = NewEIP155Signer(config.ChainID)
+	case config.IsHomestead(blockNumber):
+		signer = HomesteadSigner{}
```

Comment:
Do we want to support `HomesteadSigner` and `FrontierSigner`? If so, why?

Reply:
Is the reason for supporting those is just minimal diff? I'd presume that's why

Reply:
The only reason why they are there is for minimal diff. I'd prefer to just:
```golang
reutnr newEIP155Signer(config.ChainID)
```

Reply:
Ok yea I think it's better to make the diff here

Reply:
IMO we should always be striving for a minimal diff, even in places where it seems trivial. There's a lot of dead code which we're not using (e.g. p2p stuff, ethash etc.), but we haven't removed it, so we should not remove this part either, and try to stay as close to geth as possible.